sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -151,17 +155,58 @@ class annotated_ref<T, detail::properties_t<Props...>> {
     return *this;
   }
 
-  PROPAGATE_OP(+=)
-  PROPAGATE_OP(-=)
-  PROPAGATE_OP(*=)
-  PROPAGATE_OP(/=)
-  PROPAGATE_OP(%=)
-  PROPAGATE_OP(^=)
-  PROPAGATE_OP(&=)
-  PROPAGATE_OP(|=)
+  this_t &operator++() {
+    T &t = annotated_load();
+    ++t;
+    return *this;
+  }
+
+  this_t &operator++(int) {
+    T &t = annotated_load();
+    t++;
+    return *this;
```

Comment:
The post-increment operator should return the (reference to the) original value not the updated value.